src/index.tsx

```diff
@@ -131,29 +131,23 @@ const VirtualDragList = defineComponent({
       }
     );
 
-    watch(
-      () => virtualAttributes,
-      (newVal, oldVal) => {
-        if (!virtual) return;
-        for (let key in newVal) {
-          if (newVal[key] != oldVal[key]) {
-            virtual.option(key as any, newVal[key]);
-          }
+    watch(virtualAttributes, (newVal, oldVal) => {
+      if (!virtual) return;
+      for (let key in newVal) {
+        if (newVal[key] != oldVal[key]) {
+          virtual.option(key as any, newVal[key]);
         }
       }
-    );
+    });
```

Comment:
I have changed it to direct property access instead of getter fucntion with `deep=true` because with getter it had same property values in `newVal` and `oldVal`. But with direct property it see old and previous props correctly